lib/meadow/pipeline.ex

```diff
@@ -31,4 +42,24 @@ defmodule Meadow.Pipeline do
       initial_action.send_message(%{file_set_id: file_set_id}, context)
     end
   end
+
+  defp wait_for_checksum_tags(%{core_metadata: %{location: location}} = attrs) do
+    with %{host: bucket, path: "/" <> key} <- URI.parse(location) do
+      case wait(AWS.check_object_tags!(bucket, key, Config.required_checksum_tags()),
```

Comment:
Looking over the `wait_for_it` docs it looks like there's a convenience method [case_wait/3](https://hexdocs.pm/wait_for_it/WaitForIt.html#case_wait/3) that we might want to consider using here. I was thinking the bit:

```
The difference with case_wait is that if none of the clauses initially matches it will wait and periodically re-evaluate the clauses until one of them does match or a timeout occurs.
``` Happy to let it go but since we're using the library anyways it might be worthwhile. Thoughts?

Reply:
We had `case_wait` in there but got frustrated with it for some reason, which I can't remember now. If you don't feel strongly I vote to leave as is.

Reply:
The difference between `case wait` and `case_wait` is that with the former, you can catch `{:timeout, timeout}` inside the same block as your `{:ok, return_value}` success clauses, while with `case_wait` you put your bare `return_value` clauses inside the `case_wait` and catch the `{:timeout, timeout}` outside of it. So you'd have to wrap `case_wait` in another `case` or an `if` block, which I think is super ugly.

Reply:
Works for me!